lib/Document/StaticPageGenerator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Document;
+
+
+use Pimcore\Document\Renderer\DocumentRenderer;
+use Pimcore\Logger;
+use Pimcore\Model\Document;
+use Pimcore\Tool\Storage;
+use Symfony\Component\Lock\LockFactory;
+
+class StaticPageGenerator
+{
+    /**
+     * @var DocumentRenderer
+     */
+    protected $documentRenderer;
+
+    /**
+     * @var LockFactory
+     */
+    private $lockFactory;
+
+
+    public function __construct(DocumentRenderer $documentRenderer, LockFactory $lockFactory)
+    {
+        $this->documentRenderer = $documentRenderer;
+        $this->lockFactory = $lockFactory;
+    }
+
+    /**
+     * @param Document\PageSnippet $document
+     *
+     * @return string
+     */
+    public function getStoragePath($document)
+    {
+        return $document->getRealFullPath() . '.html';
+    }
+
+    /**
+     * @param Document\PageSnippet $document
+     *
+     * @return bool
+     */
+    public function generate($document)
+    {
+        $storagePath = $this->getStoragePath($document);
+
+        $storage = Storage::get('document_static');
+        $startTime = microtime(true);
+
+        $lockKey = 'document_static_' . $document->getId() . '_' . md5($storagePath);
+        $lock = $this->lockFactory->createLock($lockKey);
+
+        $lock->acquire(true);
+
+        try {
+            $renderedDocumentData = $this->documentRenderer->render($document);
```

Comment:
```suggestion
            $renderedDocumentData = $this->documentRenderer->render($document, ['static_page_generator' => true]);
```

Reply:
I think that would be nice, so can react to static page generation

Reply:
👍 thanks!